File: Selenium/Project/ProjectActivity7.py
from selenium import webdriver
from selenium.webdriver.firefox.service import Service as FirefoxService
from webdriver_manager.firefox import GeckoDriverManager
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait

# Set up the Firefox Driver with WebDriverManger
service = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service) as driver:

    # Open the link in browser
    driver.get("https://alchemy.hguy.co/jobs")

    # Initialize wait object
    wait = WebDriverWait(driver, 30);

    # Enter the details for new job creation
    driver.find_element(By.ID, "menu-item-26").click()
    driver.find_element(By.ID, "create_account_email").send_keys("testjob78@gmail.com")
    driver.find_element(By.ID, "job_title").send_keys("Junior Consultant")
    driver.find_element(By.ID, "job_location").send_keys("Bangalore")
    driver.find_element(By.ID, "job_description_ifr").send_keys("Test Job creation")
    driver.find_element(By.ID, "application").send_keys("https://lms.training-support.net")
    driver.find_element(By.ID, "company_name").send_keys("ABC Marketing Ltd")
    driver.find_element(By.ID, "company_website").send_keys("https://alchemy.hguy.co")
    driver.find_element(By.ID, "company_tagline").send_keys("Test")
    driver.find_element(By.ID, "company_video").send_keys("https://www.youtube.com/watch?v=QM2-MX1Lz-A")
    driver.find_element(By.ID, "company_twitter").send_keys("@Twitter")
    driver.implicitly_wait(20)
    driver.find_element(By.XPATH, "//input[@value='Preview']").click()
    driver.implicitly_wait(20)
    driver.find_element(By.XPATH, "//input[@value='Submit Listing']").click()
    driver.implicitly_wait(20)

    # Validate if Job created successfully
    driver.find_element(By.ID, "menu-item-24").click()
    driver.find_element(By.ID, "search_keywords").send_keys("ABC Marketing Ltd")
    driver.find_element(By.XPATH, "//input[@value='Search Jobs']").click()
    driver.implicitly_wait(20)
    jobName = driver.find_element(By.CLASS_NAME, "position").text
    assert jobName == "Junior Consultant\nABC Marketing Ltd Test"

    # Close the browser
    driver.quit()

File: Selenium/Project/ProjectActivity7.java
package ProjectActivities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;
import org.testng.annotations.BeforeClass;
import org.testng.annotations.Test;
import org.openqa.selenium.WebElement;

import java.time.Duration;

public class ProjectActivity7 {
    WebDriver driver;
    WebDriverWait wait;

    @BeforeClass
    public void setUp() {
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        driver = new FirefoxDriver();
        wait = new WebDriverWait(driver, Duration.ofSeconds(10));
        driver.get("https://alchemy.hguy.co/jobs/");
    }

    @Test
    public void createJobTest() throws InterruptedException {
        driver.findElement(By.id("menu-item-26")).click();
        driver.findElement(By.id("create_account_email")).sendKeys("testjob58@gmail.com");
        driver.findElement(By.id("job_title")).sendKeys("System Engineer");
        driver.findElement(By.id("job_location")).sendKeys("Bangalore");
        driver.findElement(By.id("job_description_ifr")).sendKeys(" Test Job creation");
        driver.findElement(By.id("application")).sendKeys("https://lms.training-support.net");
        driver.findElement(By.id("company_name")).sendKeys("ABS Pvt Ltd");
        driver.findElement(By.id("company_website")).sendKeys("https://alchemy.hguy.co");
        driver.findElement(By.id("company_tagline")).sendKeys("Problem One Solution");
        driver.findElement(By.id("company_video")).sendKeys("https://www.youtube.com/watch?v=QM2-MX1Lz-A");
        driver.findElement(By.id("company_twitter")).sendKeys("@Twitter");
        Thread.sleep(2000);
        driver.findElement(By.xpath("//input[@value='Preview']")).click();
        Thread.sleep(2000);
        driver.findElement(By.xpath("//input[@value='Submit Listing']")).click();
        wait.until(ExpectedConditions.elementToBeClickable(By.id("menu-item-24")));
        driver.findElement(By.id("menu-item-24")).click();
        driver.findElement(By.id("search_keywords")).sendKeys("ABS Pvt Ltd");
        driver.findElement(By.xpath("//input[@value='Search Jobs']")).click();
        wait.until(ExpectedConditions.elementToBeClickable(By.className("company_logo")));
        String jobName = driver.findElement(By.className("position")).getText();
        Assert.assertEquals(jobName, "System Engineer\nABS Pvt Ltd Problem One Solution");
    }


    @AfterClass
    public void tearDown() {
        // Close the browser
        driver.quit();
    }
}